.github/workflows/test.yml

```diff
@@ -15,11 +15,7 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-
-    - uses: actions/setup-node@v2
-      with:
-        node-version: '16'
```

Comment:
Setup node is needed if you use it, especially when running npm install. Because setup-node does caching for you.

Reply:
`actions/checkout@v3` will install npm; would it still do the caching? 🤔

Reply:
I believe `actions/checkout` is language agnostic. NPM is installed on the runner by default, but it will not do any caching. Recommend keeping the setup-node so that you both have control over the version that's used and benefit from caching.

Reply:
I don't think we are using any caching in this workflow, so it should be safe, right? 🤔

Reply:
The cache is implicit to `setup-node` action. It caches at the end of the workflow in the `post` step of `setup-node` as far as I know. Since you're using `npm install`, `npm` cache is automatically cached/filled when using `setup-node`.

I usually use Yarn where this makes a difference of 2s vs 60s for installing the dependencies `yarn install`. I'm assuming the same works for NPM but to be completely sure we might have to look at the difference in setup time for `npm install`

Reply:
I suppose since the install time is only 6 seconds right now it doesn't matter in this specific case though. Up to you.